### PR TITLE
Create centred dialog component

### PR DESCRIPTION
**Pull request by bcowell**
- Created another type of dialog with a backdrop like https://reakit.io/docs/dialog/#backdrop
- Updated package to include the `watch` command. Which will watch files for changes and re-run tests related to changed files `-i` flag to regenerate failed snapshots

### CODE REVIEW

File: src/Dialog/__snapshots__/Dialog.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`DialogCentreScreen content renders English copy 1`] = `<div />`;
+
+exports[`DialogCentreScreen content renders French copy 1`] = `<div />`;
```
Comment (fiffty) [inline src/Dialog/__snapshots__/Dialog.test.tsx.snap:5]: This is the snapshot, just a `<div />` haha. Let's just remove the snapshot tests altogether, it just adds more time to tests without adding any value.

File: src/Dialog/DialogCentreScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  z-index: 999;
+`;
+
+const StyledDialog = styled(Dialog)`
+  display: flex;
+  position: fixed;
+  top: 50%;
+  left: 50%;
+  -webkit-transform: translate(-50%, -50%);
+  -ms-transform: translate(-50%, -50%);
```
Comment (fiffty) [inline src/Dialog/DialogCentreScreen.tsx:35]: `styled-components` library automatically creates these vendor specific style definitions when it renders. No need to add them.

File: stories/components/Dialog/Dialog.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+| name     | type      | default | description             |
+| -------- | --------- | ------- | ----------------------- |
+| label    | string    |         | Aria-label              |
+| children | Component |         | Placeholder for input   |
+| dialog   | useDialog |         | Hook from reakit/Dialog |
+
+### Example
+
+```javascript
+import { useDialog } from "reakit/Dialog";
```
Comment (diegohaz) [inline stories/components/Dialog/Dialog.md:28]: Shouldn't it be importing from `@yuhuinc/yuhui`? 😅

File: stories/components/Dialog/Dialog.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
```
Comment (diegohaz) [inline stories/components/Dialog/Dialog.md:59]: Same here!

File: stories/components/Dialog/Dialog.md (hunk 3 above)
Comment (fiffty) [reply stories/components/Dialog/Dialog.md:28]: Thanks for the catch!

File: stories/components/Dialog/Dialog.md (hunk 4 above)
Comment (fiffty) [reply stories/components/Dialog/Dialog.md:59]: Thanks! 👍

**Pull request merged**